python/pyspark/sql/pandas/types.py

```diff
@@ -582,7 +585,12 @@ def convert_array(value: Any) -> Any:
                     return None
                 elif isinstance(value, np.ndarray):
                     # `pyarrow.Table.to_pandas` uses `np.ndarray`.
-                    return np.array([_element_conv(v) for v in value])  # type: ignore[misc]
+                    if ndarray_as_list:
```

Comment:
`_ndarray_as_list` instead of `ndarray_as_list`?

Reply:
Btw, shall we define a new function for it's `True` or `False` to reduce any overhead, similar to ones for `StructType`?